### PR TITLE
Erizo Controller refactoring to add Clients and Channels

### PR DESCRIPTION
**Pull request by jcague**
**Description**

Some code refactoring in Erizo Controller. Now we use the next models to improve code readability: 
* Stream: this one is not new, represents a published stream. 
* Channel: a basic transport channel that checks received tokens.
* Client: represents an end client connection. Stores its room, streams, etc. Handles events received from the client side.
* Room: represents a room, with clients and streams.

- [x] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

Not needed. All changes are in the backend.

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

**Comment by kekkokk** [comment]
It would be nice to catch the socket disconnection event and pass it to clients just to be able to close p2p connections.
ex. atm if i'm in a p2p room and I'm publishing to a client and this client goes away, my peerconnection remains open until it goes to a failed state. this is not a desiderable behavior since A) I'm sending udp packets for 10 seconds to no one

**Comment by jcague** [comment]
@kekkokk thanks for the heads up! will add a fix to this PR

**Comment by jcague** [comment]
@kekkokk done, can you check it please?

File: erizo_controller/common/amqper.js [hunk 1]
```diff
@@ -161,16 +157,23 @@ exports.bindBroadcast = function(id, callback) {
                 }
                 if (body.message.method && rpcPublic[body.message.method]) {
                     body.message.args.push(answer);
-                    rpcPublic[body.message.method].apply(rpcPublic, body.message.args);
+                    try {
+                      rpcPublic[body.message.method].apply(rpcPublic, body.message.args);
+                    } catch(e) {
```
Comment (aalonsog) [inline erizo_controller/common/amqper.js]: 👍

File: erizo_controller/erizoController/erizoController.js [hunk 2]
```diff
@@ -321,808 +275,150 @@ var updateMyState = function () {
...
-            }
-
-            index = socket.streams.indexOf(streamId);
-            if (index !== -1) {
-                socket.streams.splice(index, 1);
-            }
-            if (socket.room.streams[streamId]) {
-                delete socket.room.streams[streamId];
-            }
-            callback(true);
```
Comment (aalonsog) [inline erizo_controller/erizoController/erizoController.js]: I will miss these lines here... ;)

**Pull request merged**